vassal-app/src/main/java/VASSAL/build/module/chessclockcontrol/ChessClock.java

```diff
@@ -0,0 +1,826 @@
+
+  public static final String GENERIC   = "Player";
+
+  private static Color defaultColor;       // Stores default look-and-feel colors for buttons
+  private static Color defaultFontColor;
+
+  protected Color tickingBackgroundColor = new Color(255, 255, 0);  // Starting (default) colors
+  protected Color tickingFontColor       = new Color(51, 51, 51);
+  protected Color tockingFontColor       = new Color(0, 0, 0);
+
+  protected static Timer timer;    // Millisecond timer (but only actually updated every ~100ms)
+  protected long startTime = -1;   // Millisecond timer count of last timer check
+
+  protected LaunchButton timerButton;  // Button (and display) for this clock
+  protected String image;              // Icon for this clock
+  protected String side;               // Name of side/player whose time this clock represents
+  protected String buttonText;         // Button text for clock button
+  protected long elapsedTime;          // Apparent elapsed time
+  protected long verifiedTime;         // Time successfully verified through handshake
+  protected boolean clockTicking;      // Is this clock running
+
+  protected boolean tocking; // Tracks the half-second "pulse" for the font color
+
+  protected boolean instanceIsActive;  // True if this ChessClock has been fully initialized & registered (and not yet shut down)
+
+
+  public ChessClock(String side) {
+    this();
+    this.side       = side;
+    this.buttonText = side;
+    setConfigureName(side);
+    setTimerButton();
+  }
+
+
+  public ChessClock() {
+    defaultColor = UIManager.getColor ("Panel.background");      // Store our original default background color
+    defaultFontColor = UIManager.getColor ("Button.foreground"); // Store our original default font color
+
+    side       = GENERIC;
+    buttonText = side;
+
+    // In case we're adding these on the fly, let's see if we can find a "decent" player side to "be",
+    // in other words one that hasn't been assigned to any clock yet. This is a little bit tongue-in-cheek,
+    // but it's intended to be heuristic not perfect, and doesn't have to be perfect. So there.
+    PlayerRoster r = GameModule.getGameModule().getPlayerRoster();
+    ChessClockControl ctrl = ChessClockControl.getInstance();
+    if ((r != null) && (ctrl != null)) {
+      for (String s : r.getSides()) {
+        boolean okay = true;
+        for (ChessClock c : ctrl.getChessClocks()) {
+          if (c.getSide().equals(s)) {
+            okay = false;
+            break;
+          }
+        }
+        if (!okay) {
+          continue;
+        }
+        side       = s;
+        buttonText = s;
+        break;
+      }
+    }
+
+    setConfigureName(side);
+
+    image        = "";
+    elapsedTime  = 0;
+    verifiedTime = 0;
+    clockTicking = false;
+
+    // Timer is static so it gets created exactly once ever
+    if (timer == null) {
+      timer = new Timer(100, this);
+      timer.stop();
```

Comment:
Is this necessary? I thought timers didn't start on creation.

Reply:
I think I "inherited that". I'll take it out we can hope it works :)